src/ucs/sys/event_set.c

```diff
@@ -137,44 +136,36 @@ ucs_status_t ucs_event_set_del(ucs_sys_event_set_t *event_set, int event_fd)
 }
 
 ucs_status_t ucs_event_set_wait(ucs_sys_event_set_t *event_set,
-                                unsigned max_events, int timeout_ms,
+                                ucs_sys_event_set_event_t *events,
```

Comment:
why do we need to force user to pass a temporary array for events?
can't we allocate it inside the function with ucs_alloca or malloc? or event call epoll_wait several times?

Reply:
> why do we need to force user to pass a temporary array for events?

I agree this is a bad approach

> can't we allocate it inside the function with ucs_alloca or malloc? or event call epoll_wait several times?

now the `epoll_events` array is allocated using `ucs_alloca`